ESPResSo/poiseuille.py

```diff
@@ -71,27 +73,27 @@ def poiseuille_flow(x, force_density, dynamic_viscosity, height):
               tau=TIME_STEP, ext_force_density=FORCE_DENSITY)
 system.actors.add(lbf)
 top_wall = espressomd.shapes.Wall(normal=[1, 0, 0], dist=WALL_OFFSET)
-bottom_wall = espressomd.shapes.Wall(normal=[-1, 0, 0], dist=-(BOX_L - WALL_OFFSET))
+bottom_wall = espressomd.shapes.Wall(normal=[-1, 0, 0], dist=-(BOX_L_X - WALL_OFFSET))
 top_boundary = espressomd.lbboundaries.LBBoundary(shape=top_wall)
 bottom_boundary = espressomd.lbboundaries.LBBoundary(shape=bottom_wall)
 system.lbboundaries.add(top_boundary)
 system.lbboundaries.add(bottom_boundary)
 
 sim_xdata = (np.arange(lbf.shape[0]) + 0.5) * AGRID
 sim_ydata_list = []
-for i in tqdm.trange(30):
-    system.integrator.run(50)
+for i in tqdm.trange(30, mininterval=0.001):
+    system.integrator.run(1100)
     if (i + 1) % 10 == 0:
         fluid_velocities = (lbf[:,:,:].velocity)[:,:,:,1]
```

Comment:
Note to future maintainers: line 87 is a bottleneck for the GPU implementation. The entire data is copied from device memory to host memory and the reduction is carried out in host memory. Therefore, increasing the problem size with a larger simulation box negatively impacts the GPU implementation. This is also why the progress bar momentarily freezes at 30%, 63% and 97% with the GPU implementation. To avoid skewing the results, we increased the number of simulation steps, and adjusted the time step accordingly to recover the same curves as in the original version of the script.

Runtime on AMD Ryzen 9 9950X with Nvidia RTX 3070: 4 seconds on GPU, 1min15s on CPU.

Reply:
Perhaps best to add this as a comment? I don't think someone in the future is likely to look at merged PRs